## Importing the data and reading the data

In [26]:
import csv
import datetime

f = open("guns.csv", "r")
data = list(csv.reader(f))
print(data[0:5])

[['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education'], ['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4']]


The data contains headers. We will remove the headers for the next steps.

In [27]:
headers = data[0]
data=data[1:]
print("Headers: ")
print(headers)
print("Data: ")
print(data[0:5])

Headers: 
['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education']
Data: 
[['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4'], ['5', '2012', '02', 'Suicide', '0', 'M', '31', 'White', '100', 'Other specified', '2']]


To find relations we will use dictionaries and datetime objects to analyse the deaths by year, month, gender and race. 

## Analysing deaths by year and month

In [28]:
year_counts = {}
year= [row[1] for row in data]

for row in year:
    if row in year_counts:
        year_counts[row] += 1
    else:
        year_counts[row] = 1

print(year_counts) 

{'2012': 33563, '2014': 33599, '2013': 33636}


In [29]:
import datetime

date = [datetime.datetime(year = int(row[1]), month = int(row[2]), day = 1) 
        for row in data]

date_counts = {}

for row in date:
    if row in date_counts:
        date_counts[row] += 1
    else:
        date_counts[row] = 1

date_counts  


{datetime.datetime(2012, 1, 1, 0, 0): 2758,
 datetime.datetime(2012, 2, 1, 0, 0): 2357,
 datetime.datetime(2012, 3, 1, 0, 0): 2743,
 datetime.datetime(2012, 4, 1, 0, 0): 2795,
 datetime.datetime(2012, 5, 1, 0, 0): 2999,
 datetime.datetime(2012, 6, 1, 0, 0): 2826,
 datetime.datetime(2012, 7, 1, 0, 0): 3026,
 datetime.datetime(2012, 8, 1, 0, 0): 2954,
 datetime.datetime(2012, 9, 1, 0, 0): 2852,
 datetime.datetime(2012, 10, 1, 0, 0): 2733,
 datetime.datetime(2012, 11, 1, 0, 0): 2729,
 datetime.datetime(2012, 12, 1, 0, 0): 2791,
 datetime.datetime(2013, 1, 1, 0, 0): 2864,
 datetime.datetime(2013, 2, 1, 0, 0): 2375,
 datetime.datetime(2013, 3, 1, 0, 0): 2862,
 datetime.datetime(2013, 4, 1, 0, 0): 2798,
 datetime.datetime(2013, 5, 1, 0, 0): 2806,
 datetime.datetime(2013, 6, 1, 0, 0): 2920,
 datetime.datetime(2013, 7, 1, 0, 0): 3079,
 datetime.datetime(2013, 8, 1, 0, 0): 2859,
 datetime.datetime(2013, 9, 1, 0, 0): 2742,
 datetime.datetime(2013, 10, 1, 0, 0): 2808,
 datetime.datetime(2013, 11,

## Analysing deaths by gender and race

In [30]:
sex_list = [row[5] for row in data]
sex_counts = {}
    
for sex in sex_list:
    if sex in sex_counts:
        sex_counts[sex] += 1
    else:
        sex_counts[sex] = 1
        
sex_counts

{'F': 14449, 'M': 86349}

In [31]:
race_list = [row[7] for row in data]
race_counts = {}
    
for race in race_list:
    if race in race_counts:
        race_counts[race] += 1
    else:
        race_counts[race] = 1
        
race_counts

{'Asian/Pacific Islander': 1326,
 'Black': 23296,
 'Hispanic': 9022,
 'Native American/Native Alaskan': 917,
 'White': 66237}

## Partial findings

Analysing the gun deaths in the US between 2012 and 2014, there is not much difference in the total deaths between years, but appears to be a minor seasonal correlation, with deaths peaking in the summer and declining in the winter.  

There is a significant difference in gender deaths, as 86% of gun deaths involve man. 

To compare races, we will need some data with the total population of each race to properly compare the total deaths numbers.

## Importing a csv containing populational data

In [32]:
import csv

f = open("census.csv", "r")
census = list(csv.reader(f))
census


[['Id',
  'Year',
  'Id',
  'Sex',
  'Id',
  'Hispanic Origin',
  'Id',
  'Id2',
  'Geography',
  'Total',
  'Race Alone - White',
  'Race Alone - Hispanic',
  'Race Alone - Black or African American',
  'Race Alone - American Indian and Alaska Native',
  'Race Alone - Asian',
  'Race Alone - Native Hawaiian and Other Pacific Islander',
  'Two or More Races'],
 ['cen42010',
  'April 1, 2010 Census',
  'totsex',
  'Both Sexes',
  'tothisp',
  'Total',
  '0100000US',
  '',
  'United States',
  '308745538',
  '197318956',
  '44618105',
  '40250635',
  '3739506',
  '15159516',
  '674625',
  '6984195']]

## Finding the total deaths by 100.000 persons

In [36]:
mapping = {
    'Asian/Pacific Islander' : int(census[1][14]) + int(census[1][15]),
    'Black' : int(census[1][12]),
    'Black' : int(census[1][12]),
    'Native American/Native Alaskan' : int(census[1][13]),
    'Hispanic' : int(census[1][11]),
    'White' : int(census[1][10])
}

race_per_hundredk = {}

for key in race_counts:
    race_per_hundredk[key] = 100000 * race_counts[key] / mapping[key]
    
race_per_hundredk

{'Asian/Pacific Islander': 8.374309664161762,
 'Black': 57.877347773519595,
 'Hispanic': 20.220491210910907,
 'Native American/Native Alaskan': 24.521955573811088,
 'White': 33.56849303419181}

In [37]:
intent = [row[3] for row in data]
race = [row[7] for row in data]

homicide_race_counts = {}

for i, race in enumerate(race_list):
    if intent[i] == "Homicide":
        if race in homicide_race_counts:
            homicide_race_counts[race] += 1
        else:
            homicide_race_counts[race] = 1

homicide_race_counts
homicide_per_hundredk = {}

for key in homicide_race_counts:
    homicide_per_hundredk[key] = 100000 * homicide_race_counts[key] / mapping[key]
    
homicide_per_hundredk    



{'Asian/Pacific Islander': 3.5303462309701548,
 'Black': 48.47128498718095,
 'Hispanic': 12.627161104219912,
 'Native American/Native Alaskan': 8.717729026240365,
 'White': 4.6356417981453335}

## Findings

Analysing the gun deaths per 100000 with each race, the biggest count is for Black (57.9), followed by White (33.6), Native American (24.5) and Hispanic (20.2). 

But if we count only the homicide gun deaths, the rates changes. In this case, the Black still has the biggest rate (48.5) but now followed by Hispanic (12.6) and Native American (8.7). The White is now the second-low rate, with 4.6. This indicates that are some socio-economicals effects that affect more the minorities than the White majority. 